objects.md

```diff
@@ -0,0 +1,118 @@
+# About
+
+This repo needs to represent the following information:
+
+1. Adding/removing versions to channels
+2. Remove or add edges to all channels
+3. Remove or add edges to a specific channel
+4. Implement an "upgrade policy" object specifically for y-streams
+
+Remember that cininnatti should pull most edges from the release manifest. This data should typiclly remove edges, not add them.
+
+# Directory Structure
+
+We do this via a combination of objects and directory structure. All object should be YAML (so we can make comments). The directories in the repo should be:
+- /nodes/
+- /channels/
+- /added_edges/
+- /blocked_edges/
+- /y-policies/
+
+# Nodes
+
+The /nodes objects are very simple. It is not strictly necessary as long as cincinnatti pulls all releases from quay. However it is useful for a CI tool to build a final (smaller) graph and for disconnected customers to build a complete cincinnatti graph. (though the later is not the first target goal!) So /nodes/4.1.24 would look like:
+```
+version: 4.1.24
+payload: quay.io/openshift-release-dev/ocp-release@sha256:6f87fb66dfa907db03981e69474ea3069deab66358c18d965f6331bd727ff23f
+```
+
+# Channels
+
+The /channels object described the nodes in a channel. For example the /channels/stable-4.1.yaml file would contain
+
+```
+name: stable-4.1
+versions:
+  - 4.1.0
+  - 4.1.1
+  - ...
+  - 4.1.11
+  # 4.1.12 Was excluded from stable!!!
```

Comment:
I don't think we want comments like this.  The set of excluded nodes can be calculated by comparing available nodes with the explicitly added nodes and noticing that (for example) 4.1.12 was not in stable-4.1.  If we want to record information about explicit exclusion decisions, I'd rather close the "add 4.112 to stable-4.1" PR with a message about why we were rejecting the release, instead of landing a comment about why we were rejecting the release.

Reply:
Comments like this should/would come from architects to make it 'easy' for a human to know that a version missing is explicit. I believe such comments are absolutely non-functional non-required, but we agreed we would use them if we were still doing this 'by hand'